### ISSUE TITLE
[runtime-env] investigate why latest virtualenv breaks test_runtime_env_complicated

### ISSUE THREAD

**Issue by scv119** [ray-project/ray#34658]
### What happened + What you expected to happen

we fixed it by pin virtualenv in https://github.com/ray-project/ray/pull/34650, but root cause not identified yet

### Versions / Dependencies

latest

### Reproduction script

n/a

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by rickyyx** [comment]
https://github.com/ray-project/ray/pull/34898

**Comment by rickyyx** [comment]
Looks like it's an issue with 23.1.* , asking there https://github.com/pypa/virtualenv/issues/2568

**Comment by rkooo567** [comment]
cc @rickyyx do you know if there was a fix in?

**Comment by rickyyx** [comment]
I don't think so. I think it might be related to https://github.com/ray-project/ray/issues/34655 also.
